privmsg.go

```diff
@@ -255,12 +256,14 @@ func (robo *Robot) learn(ctx context.Context, log *slog.Logger, ch *channel.Chan
 		log.DebugContext(ctx, "no learn tag")
 		return
 	}
+	start := time.Now()
 	user := hasher.Hash(new(userhash.Hash), msg.Sender, msg.To, msg.Time())
 	if err := brain.Learn(ctx, robo.brain, ch.Learn, msg.ID, *user, msg.Time(), brain.Tokens(nil, msg.Text)); err != nil {
 		log.ErrorContext(ctx, "failed to learn", slog.Any("err", err))
 		return
 	}
-	learnedCount.Inc()
+	robo.Metrics.LearnedCount.Observe(1)
+	robo.Metrics.LearnLatency.Observe(time.Since(start).Seconds(), ch.Name)
```

Comment:
Observe (more precisely time.Since) should move up so it measures only the latency of the database-y operation.